### ISSUE TITLE
Ray Tune - YOLO ignoring lr0 and momentum because optimizer is set to auto

### ISSUE THREAD

**Issue by Fikri645** [ultralytics/ultralytics#17444]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello! I'm doing hyperparameter tuning for YOLOv11 with Ray Tune.
Here's my code:
![image](https://github.com/user-attachments/assets/3352b709-b957-4c54-b4bc-5e9d36cc2826)

Because the Optimizer is set to auto, the lr0, momentum set by Ray Tune is ignored, and using optimizer: AdamW(lr=0.002, momentum=0.9) instead.

![image](https://github.com/user-attachments/assets/a481385f-1c19-48cd-beed-9dcd689140a5)

Is it normal to be like this? or do I need to set the optimizer too?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Fikri645, thank you for reaching out to Ultralytics 🚀! We appreciate your involvement and work with YOLOv11.

If this is a 🐛 Bug Report, please ensure you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us diagnose and resolve the issue more effectively.

If your issue is related to custom training ❓ or hyperparameter tuning, please include detailed information, such as dataset samples and training logs, and confirm that you are adhering to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To further collaborate with the Ultralytics community:
- For real-time troubleshooting, join our [Discord](https://ultralytics.com/discord) 🎧.
- For more comprehensive discussions, participate on our [Discourse](https://community.ultralytics.com).
- Share experiences and discuss topics on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

We recommend upgrading to the latest `ultralytics` package, ensuring you have all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/), to check if this resolves your issue:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in the following verified environments, complete with all dependencies like [CUDA](https://developer.nvidia.com/cuda), [CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Consult our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Check the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green badge indicates that all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. These tests ensure that all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) are working correctly across various operating systems.

This is an automated response, but rest assured, an Ultralytics engineer will assist you shortly. 😃

**Comment by Y-T-G** [comment]
You have to set optimizer

**Comment by Fikri645** [comment]
Thanks for the answer sir

**Comment by glenn-jocher** [comment]
@Fikri645 you're welcome! If you have any more questions, feel free to ask.
